### PR TITLE
Implement the `UnixEpoch` function

### PR DESCRIPTION
**Pull request by rajivharlalka**
Adds the UnixEpoch function. #158  (without modifiers)

- [x] Add support for unixepoch of custom datetime

### CODE REVIEW

**Comment by rajivharlalka** [comment]
@penberg I was quite unsure of how I should add tests for this function.

**Comment by penberg** [comment]
@rajivharlalka You can test the function with a modifier:

```
sqlite> SELECT unixepoch('2024-08-05');
1722816000
```

Of course, you won't be able to cover all of the functionality in `unixepoch()` but that's OK.

**Comment by sonhmai** [comment]
hi @rajivharlalka, maybe [this PR 252](https://github.com/penberg/limbo/commit/e6985c1380ca97527f89433e08aa579a05f968aa#diff-a046d58ab24eee8207f0ce3199f8d0a609edcef9c24b8ed7f242f7a60e6c1e61) can serve as an example. 

The test can be added to `testing/scalar-functions.test` and executed with `make test`.

The tests from `sqlite` can be referenced https://github.com/sqlite/sqlite/blob/f2b21a5f57e1a1db1a286c42af40563077635c3d/test/date3.test#L36 and https://github.com/sqlite/sqlite/blob/f2b21a5f57e1a1db1a286c42af40563077635c3d/test/date.test#L604

Hope it helps!

**Comment by rajivharlalka** [comment]
@sonhmai that looks quite a neat approach for testing the function. Would implement it in some time. Thanks for the reference.

**Comment by penberg** [comment]
Hey @rajivharlalka! Are you still working on this?

**Comment by rajivharlalka** [comment]
Hi @penberg ,  Sorry had been a little drifted from this PR. would try to complete it asap as only a modifier needs to be added.

**Comment by rajivharlalka** [comment]
@penberg I feel that the modifiers are quite common among all time functions and should be taken care seperatly. Can review and merge this PR for now.

File: testing/scalar-functions.test
Comment (sonhmai) [inline testing/scalar-functions.test:1]: could you rebase main and move these tests to `testing/scalar-functions-datetime.test`? 
the tests of datetime functions are now moved to this file.

**Review by sonhmai — CHANGES_REQUESTED** [review]

**Comment by sonhmai** [comment]
@rajivharlalka can we also change `COMPAT.md` file to note that `unixepoch` function is now partially supported?

as a side note, we added modifiers in https://github.com/penberg/limbo/pull/305 for `time(..)` function, integrations with other datetime functions incl. `unixepoch` is likely similar.

**Comment by penberg** [comment]
Hey @rajivharlalka, would you be able to move the tests to the new file as pointed out by @sonhmai?

**Comment by rajivharlalka** [comment]
sorry @sonhmai , somehow missed the notification about your review. Have moved the test into the `*-datetime` test file. 
Regarding the modifiers, I will try to take it up in another PR if it would be possible for me, don't want to stall this one as it's already been long.

**Pull request closed without merging**